src/framework/components/model/component.js

```diff
@@ -621,6 +622,16 @@ Object.assign(pc, function () {
                 if (this._clonedModel) {
                     this._model.destroy();
                     this._clonedModel = false;
+                } else if (this._model) {
+                    for (i = 0; i < this._model.meshInstances.length; i++) {
+                        var mesh = this._model.meshInstances[i];
+                        if (! mesh._material) continue;
+
+                        var ind = mesh._material.meshInstances.indexOf(mesh);
+                        if (ind === -1) continue;
+
+                        mesh._material.meshInstances.splice(i, 1);
```

Comment:
Why don't you just do:

```
mesh.material = null;
```

Reply:
Also, it should be `meshInstance`, not `mesh`.

Reply:
Updated it.